src/redis-cli.c

```diff
@@ -4704,6 +4705,41 @@ static clusterManagerNode *clusterManagerGetSlotOwner(clusterManagerNode *n,
     return owner;
 }
 
+static void clusterManagerSetSlotNodeReplicaOnly(clusterManagerNode *node1,
+                                                 clusterManagerNode *node2,
+                                                 int slot) {
+    /* A new CLUSTER SETSLOT variant that finalizes slot
+     * ownership on replicas only (CLUSTER SETSLOT s 
+     * NODE n REPLICAONLY) is introduced in Redis 7.2+
+     * to help mitigate the single-point-of-failure
+     * issue related to the slot ownership finalization on
+     * HA clusters. We make a best-effort attempt below to
+     * utilize this enhanced reliability. Regardless of the
+     * result, we continue with finalizing slot ownership
+     * on the primary nodes. Note that this command is not
+     * essential. Redis 7.2+ will attempt to recover from
+     * failed slot ownership finalizations if they occur,
+     * although there may be a brief period where slots
+     * caught in this transition stage are unavailable.
+     * Including this additional step ensures no downtime
+     * for these slots if any failures arise. */
+    redisReply *reply = CLUSTER_MANAGER_COMMAND(node1, "CLUSTER SETSLOT %d NODE %s REPLICAONLY",
+                                                slot, (char *) node2->name);
+    if (reply->type == REDIS_REPLY_ERROR) {
+        /* Either target Redis server doesn't support this command or the slot is
+         * not in the right state*/
+        clusterManagerLogWarn("*** Failed to finalize slot on replicas: %s\n", reply->str);
+        freeReplyObject(reply);
+        return;
+    } 
+    clusterManagerLogInfo(">>> Waiting for at least one replica to complete the slot finalization\n");
+    reply = CLUSTER_MANAGER_COMMAND(node1, "WAIT 1 1000");
```

Comment:
Why exactly 1? We don't know the number of replicas, do we? What if there are no replicas at all?

What if the primary crashes when one replica has replicated and the other replicas haven't, is it likely that this replica will be promoted because it has a better replica rank than the others?

That's what I can think of. What's your reasoning?

Reply:
>Why exactly 1? We don't know the number of replicas, do we? 

We do know the number of replicas though I was thinking that having one replica acknowledging is sufficient since we would need to lose both the primary and the replica in question to hit the original SPOF issue. On a second thought though, I am now convinced that we should wait for all replicas since the intent is "best effort only" anyways. Will send in an update next.

>What if there are no replicas at all?

I have modified the engine to return a new error string "Target node has no replica".

>What if the primary crashes when one replica has replicated and the other replicas haven't, is it likely that this replica will be promoted because it has a better replica rank than the others?

Yes, intuitively the replica who ack'd the replication would rank higher compared to the ones who didn't. That said, will wait for all replicas next.